core/drivers/stm32_tamp.c

```diff
@@ -98,46 +111,263 @@
 #define _TAMP_VERR_MAJREV		GENMASK_32(7, 4)
 
 /*
- * TAMP instance data
- * @base - IOMEM base address
- * @clock - TAMP clock
- * @it - TAMP interrupt number
- * @hwconf1 - Copy of TAMP HWCONF1 register content
- * @hwconf2 - Copy of TAMP HWCONF2 register content
- * @compat - Reference to compat data passed at driver initialization
+ * CIDCFGR register bitfields
  */
-struct stm32_tamp_instance {
-	struct io_pa_va base;
-	struct clk *clock;
-	int it;
-	uint32_t hwconf1;
-	uint32_t hwconf2;
-	struct stm32_tamp_compat *compat;
-};
+#define _TAMP_CIDCFGR_SCID_MASK		GENMASK_32(6, 4)
+#define _TAMP_CIDCFGR_CONF_MASK		(_CIDCFGR_CFEN |	 \
+					 _CIDCFGR_SEMEN |	 \
+					 _TAMP_CIDCFGR_SCID_MASK)
+
+/* _TAMP_BKPRIFR */
+#define _TAMP_BKPRIFR_1_MASK		GENMASK_32(7, 0)
+#define _TAMP_BKPRIFR_2_MASK		GENMASK_32(7, 0)
+#define _TAMP_BKPRIFR_3_MASK		(GENMASK_32(23, 16) | GENMASK_32(7, 0))
+#define _TAMP_BKPRIFR_ZONE3_RIF2_SHIFT	16U
+
+/*
+ * RIF miscellaneous
+ */
+#define TAMP_RIF_RESOURCES		3U
+#define TAMP_RIF_OFFSET_CNT		4U
 
 /*
  * Compatibility capabilities
- * TAMP_HAS_REGISTER_SECCFG - Supports SECCFGR, otherwise supports SMCR register
+ * TAMP_HAS_REGISTER_SECCFGR - Supports SECCFGR, otherwise supports SMCR
+ * register
  * TAMP_HAS_REGISTER_PRIVCFG - Supports PRIVCFGR configuration register
+ * TAMP_HAS_RIF_SUPPORT - Supports RIF
  */
-#define TAMP_HAS_REGISTER_SECCFG	BIT(0)
+#define TAMP_HAS_REGISTER_SECCFGR	BIT(0)
 #define TAMP_HAS_REGISTER_PRIVCFGR	BIT(1)
+#define TAMP_HAS_RIF_SUPPORT		BIT(31)
 
-/*
- * @nb_monotonic_counter - Number of monotic counter supported
- * @tags - Bit flags TAMP_HAS_* for compatibily management
+/**
+ * struct stm32_tamp_compat - TAMP compatible data
+ * @nb_monotonic_counter: Number of monotic counter supported
+ * @tags: Bit flags TAMP_HAS_* for compatibility management
  */
 struct stm32_tamp_compat {
 	int nb_monotonic_counter;
 	uint32_t tags;
 };
 
+/*
+ * struct stm32_bkpregs_conf_new - Interface for stm32_tamp_set_secure_bkpregs()
```

Comment:
Replace `Interface for stm32_tamp_set_secure_bkpregs` with something like
`Backup registers zone bounds`

Reply:
Good suggestion